backupccl: add 19.2 version gate for partitioned backup

thoszhang opened the pull request:
This PR adds a new cluster version for partitioned backup, and adds a check
that a mixed-version cluster whose upgrade to 19.2 has not been finalized
cannot create partitioned backups.

Release justification: This is a small fix for a new feature.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39374)
<!-- Reviewable:end -->

dt reviewed (approved):
I think I mentioned this offline, but over on IMPORT I've been holding off on adding until we've let it bake a bit, just in case there are any more proto changes needed in followups.

lucy-zhang committed:
backupccl: add 19.2 version gate for partitioned backup

This PR adds a new cluster version for partitioned backup, and adds a check
that a mixed-version cluster whose upgrade to 19.2 has not been finalized
cannot create partitioned backups.

Release justification: This is a small fix for a new feature.

Release note: None

lucy-zhang commented:
bors r+
